db/migrate/20170705124020_move_order_to_graders.rb

```diff
@@ -0,0 +1,32 @@
+class MoveOrderToGraders < ActiveRecord::Migration[5.1]
+  class AssignmentGrader < ApplicationRecord
+  end
+  def up
+    add_column :graders, :order, :integer
+    add_column :graders, :assignment_id, :integer
+    AssignmentGrader.all.each do |ag|
+      ag.grader.update(order: ag.order, assignment_id: ag.assignment_id)
+    end
+    Grader.where(order: nil).update(order: DateTime.now.to_i)
+    # diagnostics to find orphaned graders and grades
+    Grader.where(assignment_id: nil).each{|g| print "#{g.id} #{g.type}\n"}
+    print Grader.where(order: nil).map(&:id)
+    change_column_null :graders, :order, false
+    change_column_null :graders, :assignment_id, false
+    drop_table :assignment_graders
```

Comment:
This wasn't my preferred solution (since ordering doesn't feel intrinsic to grader-hood), but rails doesn't properly support attributes on the join-table itself in any programmatically useful way. :-\